app/service/rest.py

```diff
@@ -250,6 +252,7 @@ def update_service(service_id):
     fetched_service = dao_fetch_service_by_id(service_id)
     # Capture the status change here as Marshmallow changes this later
     service_going_live = fetched_service.restricted and not req_json.get('restricted', True)
+    message_limit_changed = fetched_service.message_limit != req_json.get('message_limit', None)
```

Comment:
are there situations where 'message_limit' is never passed down?
in those situations
`fetched_service.message_limit` will be some value while req_json.get('message_limit', None) will return `None`
In that case the cache will be deleted.

Suggestion:
`message_limit_changed = fetched_service.message_limit != req_json.get('message_limit') if 'message_limit' in req_json else False`

Reply:
Right, forgot this case. Thanks for catching it! Sorry about that.